pkg/virtctl/memorydump/memorydump.go

```diff
@@ -0,0 +1,365 @@
+)
+
+var (
+	claimName    string
+	createClaim  bool
+	storageClass string
+	accessMode   string
+	outputFile   string
+)
+
+type command struct {
+	clientConfig clientcmd.ClientConfig
+}
+
+type memoryDumpCompleteFunc func(kubecli.KubevirtClient, string, string, time.Duration, time.Duration) (string, error)
+
+// WaitMemoryDumpCompleted is used to store the function to wait for the memory dump to be complete.
+// Useful for unit tests.
+var WaitMemoryDumpComplete memoryDumpCompleteFunc = waitForMemoryDump
+
+func usageMemoryDump() string {
+	usage := `  #Dump memory of a virtual machine instance called 'myvm' to an existing pvc called 'memoryvolume'.
+  {{ProgramName}} memory-dump get myvm --claim-name=memoryvolume
+
+  #Create a PVC called 'memoryvolume' and dump the memory of a virtual machine instance called 'myvm' to it.
+  {{ProgramName}} memory-dump get myvm --claim-name=memoryvolume --create-claim
+
+  #Create and download memory dump to the given output file.
+  {{ProgramName}} memory-dump get myvm --claim-name=memoryvolume --create-claim --output=memoryDump.dump.gz
+
+  #Dump memory again to the same virtual machine with an already associated pvc(existing memory dump on vm status).
+  {{ProgramName}} memory-dump get myvm
+
+  #Download the last memory dump associated on the vm 'myvm' to the given output file.
+  {{ProgramName}} memory-dump download myvm --output=memoryDump.dump.gz
+
+  #Remove the association of the memory dump pvc (to be able to dump to another pvc).
+  {{ProgramName}} memory-dump remove myvm
+  `
+	return usage
+}
+
+// NewMemoryDumpCommand returns a cobra.Command to handle the memory dump process
+func NewMemoryDumpCommand(clientConfig clientcmd.ClientConfig) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:     "memory-dump get/download/remove (VM)",
+		Short:   "Dump the memory of a running VM to a pvc",
+		Example: usageMemoryDump(),
+		Args:    templates.ExactArgs("memory-dump", 2),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			c := command{clientConfig: clientConfig}
+			return c.run(args)
+		},
+	}
+	cmd.SetUsageTemplate(templates.UsageTemplate())
+	cmd.Flags().StringVar(&claimName, claimNameArg, "", "pvc name to contain the memory dump")
+	cmd.Flags().BoolVar(&createClaim, createClaimArg, false, "Create the pvc that will conatin the memory dump")
+	cmd.Flags().StringVar(&storageClass, storageClassArg, "", "The storage class for the PVC.")
+	cmd.Flags().StringVar(&accessMode, accessModeArg, "", "The access mode for the PVC.")
+	cmd.Flags().StringVar(&outputFile, "output", "", "Specifies the output path of the memory dump to be downloaded.")
+
+	return cmd
+}
+
+func (c *command) run(args []string) error {
+	namespace, _, err := c.clientConfig.Namespace()
+	if err != nil {
+		return err
+	}
+	virtClient, err := kubecli.GetKubevirtClientFromClientConfig(c.clientConfig)
+	if err != nil {
+		return fmt.Errorf("cannot obtain KubeVirt client: %v", err)
+	}
+
+	vmName := args[1]
+	switch args[0] {
```

Comment:
Would here make sense to simply add 3 cobra subcommands to the dumpMemory instead of using the switch? You can use persistent flags to make the option also available in the subcommands